app/controllers/journeys/expenses/tailoring/AdvertisingOrMarketingController.scala

```diff
@@ -17,57 +17,64 @@
                                                   view: AdvertisingOrMarketingView)(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport {
 
-  val businessId = "SJPR05893938418" // TODO SASS-5658 replace default BID and taxYear vals
-  val taxYear    = LocalDate.now.getYear
+  def onPageLoad(taxYear: Int, businessId: String, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+      val preparedForm = request.userAnswers.get(AdvertisingOrMarketingPage, Some(businessId)) match {
+        case None        => formProvider(userType(request.user.isAgent))
+        case Some(value) => formProvider(userType(request.user.isAgent)).fill(value)
+      }
 
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) { implicit request =>
-    val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.userId)).get(AdvertisingOrMarketingPage, Some(businessId)) match {
-      case None        => formProvider(userType(request.user.isAgent))
-      case Some(value) => formProvider(userType(request.user.isAgent)).fill(value)
-    }
-
-    Ok(view(preparedForm, mode, userType(request.user.isAgent), taxYear, businessId))
+      Ok(view(preparedForm, mode, userType(request.user.isAgent), taxYear, businessId))
   }
 
-  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData) async { implicit request =>
-    formProvider(userType(request.user.isAgent))
-      .bindFromRequest()
-      .fold(
-        formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, userType(request.user.isAgent), taxYear, businessId))),
-        value =>
-          for {
-            updatedAnswers <- Future.fromTry(
-              request.userAnswers.getOrElse(UserAnswers(request.userId)).set(AdvertisingOrMarketingPage, value, Some(businessId)))
-            _ <- sessionRepository.set(updatedAnswers)
-          } yield Redirect(navigator.nextPage(AdvertisingOrMarketingPage, mode, updatedAnswers))
-      )
+  def onSubmit(taxYear: Int, businessId: String, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) async {
+    implicit request =>
+      selfEmploymentService.getAccountingType(request.user.nino, businessId, request.user.mtditid) flatMap {
+        case Left(_) => Future.successful(Redirect(JourneyRecoveryController.onPageLoad()))
+        case Right(accountingType) =>
+          formProvider(userType(request.user.isAgent))
+            .bindFromRequest()
+            .fold(
+              formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode, userType(request.user.isAgent), taxYear, businessId))),
+              value =>
+                for {
+                  updatedAnswers <- Future.fromTry(request.userAnswers.set(AdvertisingOrMarketingPage, value, Some(businessId)))
+                  _              <- sessionRepository.set(updatedAnswers)
```

Comment:
controllers shouldn't be doing a lot of things, service layer should have business logic.

Talking to controller should be done via Service, have a look here https://github.com/hmrc/income-tax-self-employment-frontend/blob/038e67469937618795c8795146b9989747800015/app/controllers/journeys/expenses/repairsandmaintenance/RepairsAndMaintenanceAmountController.scala#L75